tests/nosetests/pyanaconda_tests/modules/network/module_network_kickstart_test.py

```diff
@@ -0,0 +1,39 @@
+#
+# Copyright (C) 2021  Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+from pykickstart.errors import KickstartError
+from pyanaconda.core.kickstart.specification import KickstartSpecificationHandler, \
+    KickstartSpecificationParser
+from pyanaconda.modules.network.kickstart import NetworkKickstartSpecification
+import unittest
+
+
+class NetworkCommandTestCase(unittest.TestCase):
+    def hostname_parse_test(self):
+        """Test Network command hostname parsing"""
+        handler = KickstartSpecificationHandler(NetworkKickstartSpecification)
+        parser = KickstartSpecificationParser(handler, NetworkKickstartSpecification)
+
+        # works ok
+        parser.readKickstartFromString("network --hostname blah.blah.blah")
+        parser.readKickstartFromString("network --hostname still.good-with.hyphen")
```

Comment:
From `module_storage_test.py`:

```
        ks_in = """
        autopart --fstype=invalid
        """
        ks_out = ""
        self._test_kickstart(ks_in, ks_out, ks_valid=False)
```
Please, add a similar test to `module_network_test.py` instead of this one.